packages/gnosis-relay-api/src/tests/GnosisRelayAPI.test.ts

```diff
@@ -0,0 +1,132 @@
+const SAFE_FUNDER_PUBLIC_KEY = "0x913dA4198E6bE1D5f5E4a40D0667f70C0B5430Eb";
+
+const NULL_ADDRESS = "0x0000000000000000000000000000000000000000";
+const DUMMY_ADDRESS = "0x0000000000000000000000000000000000000003";
+
+export async function sleep(milliseconds: number): Promise<void> {
+    return new Promise<void>(resolve => setTimeout(resolve, milliseconds));
+}
+
+let api: GnosisRelayAPI;
+
+beforeAll(async () => {
+    api = new GnosisRelayAPI("https://safe-relay.rinkeby.gnosis.pm/api/");
+  }, 1000);
+
+test('Gnosis Relay API:: Make safe and do transactions', async () => {
+
+    const url = "https://eth-rinkeby.alchemyapi.io/jsonrpc/xWkVwAbM7Qr-p8j-Zu_PPwldZJKmaKjx";
+    const provider = new ethers.providers.JsonRpcProvider(url);
+    const wallet = new ethers.Wallet(SAFE_FUNDER_PRIVATE_KEY, provider);
+
+    let signingKey = new ethers.utils.SigningKey(SAFE_FUNDER_PRIVATE_KEY);
+
+    // TODO provide more params and confirm a pre-computed address is equal to the address generated
+
+    const gnosisSafeData = {
+        saltNonce: Number((Math.random() * 10000000).toFixed()),
+        threshold: 1,
+        owners: [SAFE_FUNDER_PUBLIC_KEY],
+        paymentToken: NULL_ADDRESS,
+    };
+
+    // Get safe creation data
+    console.log(`Getting Safe Creation Data`);
+    const safeResponse = await api.createSafe(gnosisSafeData);
+    const safeAddress = safeResponse.safe;
+    const payment = safeResponse.payment;
+    await expect(safeAddress).not.toEqual(NULL_ADDRESS);
+    console.log(`Expected Safe Address: ${safeAddress}`);
+
+    // The safe is not yet funded (or deployed)
+    let safeStatus = await api.checkSafe(safeAddress);
+    await expect(safeStatus.blockNumber).toEqual(null);
+    await expect(safeStatus.txHash).toEqual(null);
+
+    // Fund the safe
+    console.log(`Funding Safe`);
+    const fundingTransaction = {
+        nonce: await wallet.getTransactionCount(),
+        to: safeAddress,
+        value: new ethers.utils.BigNumber(payment),
+    };
+    await wallet.sendTransaction(fundingTransaction);
+    safeStatus = await api.checkSafe(safeAddress);
+
+    // Wait till the relay service has deployed the safe
+    console.log(`Waiting for Safe Deployment`);
+    while (safeStatus.blockNumber === null) {
+        await sleep(2000);
+        safeStatus = await api.checkSafe(safeAddress);
+    }
+
+    console.log(`Depositing Additional Funds`);
+    const depositTransaction = {
+        nonce: await wallet.getTransactionCount(),
+        to: safeAddress,
+        value: new ethers.utils.BigNumber(10).pow(16),
+    };
+    const txResponse = await wallet.sendTransaction(depositTransaction);
+    console.log(`Waiting on TX: ${txResponse.hash}`);
+    await provider.getTransactionReceipt(txResponse.hash);
+
+    // Lets send a transaction through the safe using the relay service
+    const gnosisSafe = new ethers.Contract(safeAddress, abi["GnosisSafe"], provider);
```

Comment:
This looks like its using Rinkeby? Will the account thats being used to fund the test eventually run out of rinkeby eth?

Reply:
Yea I mentioned that above. There were some issues I encountered with running a local relay that I punted on in favor of just testing on Rinkeby for now. This isnt in CI so it'll only spend funds when run manually at the moment. That said its a public private key so anyone could drain the account.